### DISCUSSION TITLE
keybinding for screen capture script doesn't work

### DISCUSSION THREAD

**Discussion by arpeasal** [i3/i3#5349] [Q&A] [answered]
I created 3 script files in my `~/.config/i3/scripts` directory.  The scripts are reboot,sh, poweroff.sh and screencapture.sh.

I made them all executable (chmod +x).

I added the keybindings below in my `~/.config/i3/config` file.
`bindsym $mod+y exec /home/randall/.config/i3/scripts/poweroff.sh`
`bindsym $mod+shift+y exec /home/randall/.config/i3/scripts/reboot.sh`
`bindsym $mod+$alt+y exec /home/randall/.config/i3/scripts/screencapture.sh`
Note: In the config file I have
set $alt Mod1

The first 2 keybindings work fine, but the screencapture binding does NOT.

I performed the tests below to try to find the problem.  After making the changes, I did each of the following to see if the changes worked.  None did.
Reloaded the config file
Restarted i3
Shut down i3 and started it again
Rebooted the laptop

First, I verified there was no keybinding conflict by doing a search of the config file for "$mod+$alt+s".  The only occurence of that was the one I made for the screencapture script, so there is no conflict.

Tested the keybinding combination by replacing the script with a program.
`bindsym $mod+$alt+y exec gimp`
Gimp started fine, so the keybinding works fine

Note: The screencapture script file has the shebang plus one other line:
`scrot /galpao/screen_capture/%Y-%m-%d-%T-screenshot.png -s`
I tested the script by running `/home/randall/.config/i3/scripts/screencapture.sh` in a terminal window.
The screen capture worked fine, so the script is fine.

In the poweroff keybinding (which was working fine), I replaced the poweroff script name with screencapture script name.  The changed keybinding no longer worked.

Whenever I try the keybinding with the screencapture script, whichever window is focused at that time flashes.

What am I missing?

**Comment by Airblader** [comment, marked as answer]
Use `bindsym --release`.

**Reply by arpeasal** [reply]
Thanks, that solved the problem.  Do you mind if I ask why that is necessary?  Nevermind.  I found an explanation in the user guide on the i3 website: (Some tools (such as import or xdotool) might be unable to run upon a KeyPress event, because the keyboard/pointer is still grabbed. For these situations, the --release flag can be used, which will execute the command after the keys have been released.)

Note: It was a one line script and I originally tried just executing the scrot (the screen capture program) command with the keybinding, but it didn't work.  I tried it again with the `--release` flag and it worked fine.  So I don't need to do the script file.   Also, since that worked I tried it with the reboot and poweroff scripts.  I changed the keybindings to include the `--release` flag and `exec systemctl reboot` and `exec systemctl poweroff`.  They worked fine, so I have deleted the whole `.config/i3/scripts` directory.

Thanks again.
